src/lib.rs

```diff
@@ -1,3 +1,37 @@
+//! This simple crate allows you to translate a *UUID* to a pokémon name. The purpose is to provide
+//! simple names to address objects, so you can easily talk with people about the objects.
+//!
+//! The function is not injective. This means several *UUID*s will give the same name. We don't
+//! consider it an issue since context (like owner of object) will help dedup the search.
+//!
+//! ## Examples
+//! ```rust
+//! extern crate uuid;
+//! extern crate uuid_to_pokemon;
+//!
+//! use uuid::Uuid;
+//! use uuid_to_pokemon::{uuid_to_pokemon, PokemonUuid};
+//! use std::fmt::Write; // needed for write!
+//!
+//! fn main() {
+//!     let my_uuid: Uuid = Uuid::nil();
+//!     let result: PokemonUuid = uuid_to_pokemon(my_uuid);
```

Comment:
yeah, you see types often in examples so the reader can understand what's going on without looking into the specific fn definition. i can remove them if you want